### PR TITLE
Adds function to find Total Force on a RigidBody

### LINKED ISSUE #16290: No function to calculate force in rigid body file
There is no function to calculate the force acting on a body in **rigidbody`** file in **mechanics** module. I wonder how such an important feature might be missing.

### PR DESCRIPTION
**Pull request by arooshiverma**
#### References to other Issues or PRs

Fixes #16290 

#### Brief description of what is fixed or changed
Added new function in RigidBody.py- totalforce(self,frame)
Added new test in test_rigidbody- test_rigidbody_totalforce()

#### Release Notes

* physics.mechanics
  * added new function to calculate total external force on the Rigid Body

### CODE REVIEW

**Comment by abhinavsagar** [comment]
@asmeurer @sylee957 It looks good. Could you please merge this if everything is fine.

**Comment by oscarbenjamin** [comment]
This PR seems to have unrelated commits in.

It seems to me that the function added here wouldn't be valid for a non-inertial frame.

**Comment by arooshiverma** [comment]
> It seems to me that the function added here wouldn't be valid for a non-inertial frame.

While calculating force for a non-inertial frame, the code will take in account the pseudo-force too. Should the output exclude the pseudo forces?

**Comment by oscarbenjamin** [comment]
I don't know what the output should have. How is this class used? In what situation is this additional function useful?

**Comment by arooshiverma** [comment]
This function calculates the total force on the body. When we have applied many external force vectors on a rigid body, the net force can be found out by multiplying the acceleration of the center of mass of the body and the mass of the body. In cases of an inertial frame, it gives the vector sum of all the forces applied.
In case of a non-inertial frame, It gives the vector sum of Pseudo Force along with the sum of all external forces on a body. Thus this function can indirectly be used to calculate the Pseudo force on a body in a non-inertial frame.

**Comment by oscarbenjamin** [comment]
The total force could be calculated from the acceleration like this or it could be calculated from knowledge of the forces. I would expect the latter. I would want the total force expression to come out in terms of force type quantities e.g. `m*g + k*x` rather than kinematic ones e.g. `m dv/dt`.

Can you show a complete example where this total force function is useful? The test you've added does not demonstrate the usefulness of this function (to me).

**Comment by arooshiverma** [comment]
Oh I get the point. I believe I wrote this function for when the expression of velocity is in terms of some other dynamic variables.I'll improvise the test written.

**Comment by arooshiverma** [comment]
For example while working on a RigidBody, we usually set the velocity to some value. If we need the force, we can calculate it only in terms of the velocity set.

In case of a Body, we use apply_force() to append new values to the list of forces. In that case we can calculate the net force by doing the vector sum of all the applied forces.

@oscarbenjamin , @abhinavsagar your thoughts on this?

**Comment by oscarbenjamin** [comment]
Can you show a full example where this is useful? It's a lot easier to decide how to do something if you can see the problem that is being solved.

**Comment by arooshiverma** [comment]
For example:
Let's take the case that we have many particles going around an unknown planet. The planet exerts a force of unknown nature on them. Now, we know the velocity vectors v of each particle as a function of time. Using total_force() we can determine the amount of force on each particle (and thus may deduce the pattern/nature of that particular force later).

**Comment by abhinavsagar** [comment]
I think @arooshiverma has given a good enough example. But I feel there should be two different functions for calculating the forces in inertial and non inertial frames depending on the use.
@oscarbenjamin What do you think?

**Comment by oscarbenjamin** [comment]
I don't think it is a good example. In the case of modelling particles going around a planet it is much more likely that the total force is known as a function of positions a priori than that we have an equation for the velocities and are working backwards to get the force via the acceleration. Of course if we are in the situation of inferring forces from accelerations then it's easy enough to calculate the requires force from F=ma (as this patch does).

The problem is that the function you are adding is almost trivial. It's not trivial because it assumes an inertial frame but adding the function does nothing to help with the non-trivial problem because it doesn't verify that the frame is inertial and doesn't even document that it should be an inertial frame.

So it's not clear to me that this function adds much and that's why I asked for an example. Probably a better example would be something involving constrained motion.

The name of the function seems misleading to me since it is called total force but really computes the time derivative of the linear momentum. Those two are equal when the frame is inertial but nothing is done to guarantee that it is inertial.

**Comment by moorepants** [comment]
I don't think this is a good addition. The RigidBody does have a notion of its motion but not any forces applied to it. If the user wants to calculate linear and angular acceleration or momentum it is very simple to do and those are already provided. For example `RigidBody.linear_momentum(Frame).dt(Frame)` gives the result you seem to be after.

File: sympy/assumptions/tests/test_refine.py [hunk 1]
```diff
@@ -144,6 +145,12 @@ def test_atan2():
     assert refine(atan2(y, x), Q.negative(y) & Q.zero(x)) == -pi/2
     assert refine(atan2(y, x), Q.zero(y) & Q.zero(x)) == nan
 
+def test_matrixelement():
```
Comment (Shekharrajak) [inline sympy/assumptions/tests/test_refine.py]: `git revert` all the matrix related changes. Learn about git branch as well.

- Always start new PR with new branch -- checkout from master branch only.
- Here you can either **revert** all the matrix related commits and force push.

Comment (arooshiverma) [reply sympy/assumptions/tests/test_refine.py]: Yes. I'll take care the next time.

**Comment by czgdp1807** [comment]
In reference to https://github.com/sympy/sympy/pull/16320#issuecomment-474951011 and inactivity, I am closing this for now. However, if someone supports re-opening it. Please let me know. 
Thanks for the contributions and effort. Apologies for inconveniences.

**Pull request closed without merging**